pandas/tests/io/excel/test_style.py

```diff
@@ -205,3 +205,21 @@ def custom_converter(css):
 
         with contextlib.closing(openpyxl.load_workbook(path)) as wb:
             assert wb["custom"].cell(2, 2).font.color.value == "00111222"
+
+
+def test_styler_to_s3(self, s3_resource, s3so):
+    import time
+
+    mock_bucket_name, target_file = "pandas-test", "test.xlsx"
+    df = DataFrame({"x": [1, 2, 3], "y": [2, 4, 6]})
+    styler = df.style.set_sticky(axis="index")
+    styler.to_excel(f"s3://{mock_bucket_name}/{target_file}", storage_options=s3so)
+    timeout = 5
+    while True:
+        if target_file in (
+            obj.key for obj in s3_resource.Bucket("pandas-test").objects.all()
+        ):
+            break
+        time.sleep(0.1)
+        timeout -= 0.1
+        assert timeout > 0, "Timed out waiting for file to appear on moto"
```

Comment:
This test looks good! Can you also load the excel file from s3 and check the values to do a complete round trip.

Reply:
The test here meant to test if storage_option works for to_excel for styler, otherwise there is a test for dataframe.to_excel from s3 in `pandas/tests/io/excel/test_readers.py` called `test_read_from_s3_object`.
Unless I misunderstood you @rhshadrach

Reply:
Agreed there is a test for DataFrame.to_excel; here you are currently testing that _something_ is written, I'm suggesting we should be checking that the correct values are written when using storage_option with styler.to_excel.